code/datums/components/suicide_count.dm

```diff
@@ -0,0 +1,67 @@
+/*
+
+	A component designed to log and count who has suicided with the attached item.
+	It can either be attached to an item like usual, or will be added automatically with NONE when someone suicides.
+	If view_mode is HOLY, then only those spiritual enough are able to detect the ghosts' spirits.
+	If it's ALL, any old folk can come and examine for the suicide information.
+	Forensics scanners can always detect an item's suicide counts. (not implemented yet todo)
+*/
+
+/datum/component/suicide_count
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/view_mode
+	var/last_person
+	var/count = 0
+	var/datum/callback/on_die
```

Comment:
What do any of these vars do, please document them instead of just marking the request as resolved without actually documenting them